app/src/main/java/net/gsantner/markor/frontend/textview/HighlightingEditor.java

```diff
@@ -109,66 +101,25 @@ public void afterTextChanged(final Editable s) {
     // ---------------------------------------------------------------------------------------------
 
     private boolean isScrollSignificant() {
-        return Math.abs(_hlRect.top - _olhHlRect.top) > _hlShiftThreshold ||
-                Math.abs(_hlRect.bottom - _olhHlRect.bottom) > _hlShiftThreshold;
-    }
-
-    private void updateDynamicHighlighting() {
-        if (_isDynamicHighlightingEnabled) {
-            updateHighlighting(false);
-        }
-    }
-
-    private synchronized void postUpdateDynamicHighlighting() {
-        if (!_hlPostQueued) {
-            _hlPostQueued = true;
-            post(() -> {
-                updateDynamicHighlighting();
-                _hlPostQueued = false;
-            });
-        }
+        return ( _oldHlRect.top - _hlRect.top) > _hlShiftThreshold ||
+                (_hlRect.bottom -  _oldHlRect.bottom) > _hlShiftThreshold;
     }
 
     private void updateHighlighting(final boolean recompute) {
-        final Layout layout;
-        if (!_isUpdatingDynamicHighlighting && _hlEnabled && _hl != null && (layout = getLayout()) != null) {
-            _isUpdatingDynamicHighlighting = true;
+        if (_hlEnabled && _hl != null && getLayout() != null) {
 
             final boolean visible = getLocalVisibleRect(_hlRect);
 
             // Don't highlight unless shifted sufficiently or a recompute is required
             if (recompute || (visible && _hl.hasSpans() && isScrollSignificant())) {
 
-                // Addition of spans which require reflow can shift text on re-application of spans
-                // we compute the resulting shift and scroll the view to compensate in order to make
-                // the experience smooth for the user
-                final int shiftTestLine = layout.getLineForVertical(_hlRect.centerY());
-                final int oldOffset = layout.getLineBaseline(shiftTestLine);
-
                 final int[] newHlRegion = hlRegion(_hlRect); // Compute this _before_ clear
-                try {
-                    beginBatchEdit();
-                    blockBringPointIntoView(); // Hack to block bring point into view
```

Comment:
@gsantner 

After trying to tweak and optimize this for weeks (months?) now, I have given up. Dynamically adding `UpdateLayout` spans causes a million problems in `TextViews` and I would rather work on actually useful features.

I have therefore reverted to a suggestion you made when dynamic highlighting was initially being developed - separate the UpdateLayout spans and apply them statically (i.e. to the whole file) and apply only spans which do not update layout dynamically.

In practice this means that
- TodoTxt (and other formats, but mostly todotxt) will be much less janky
- Larger TodoTxt files will become slower to edit.

On my pixel 3, a 1000 line todo.txt file is definitely slow to edit, 500 lines works fine. I think this is ok for 99.9% of users. (Markdown etc files are essentially unaffected.)

As a side affect, the code is dramatically simpler

Reply:
Hm just some feedback, on git master it seems to work quite well. Not sure how much different way you tried here exactly, but that one looks pretty good to me :-).

Reply:
Yeah. It works well enough 80% of the time (after so many layers of hacks and tweaking). But at least once per editing session I will have the cursor scroll off the screen or the editor will not scroll correctly to the correct line etc etc. It makes working with todo.txt in markor very annoying and breaks my flow when I am trying to plan things.

In practice, I think this new (old) approach is a better compromise - much simpler system and stable system with somewhat lower performance.

Reply:
do you need some testfiles again or do you have it all around?